libr/core/linux_heap_glibc.c

```diff
@@ -303,20 +334,42 @@ static void GH(print_arena_stats)(RCore *core, GHT m_arena, MallocState *main_ar
 }
 
 static bool GH(r_resolve_main_arena)(RCore *core, GHT *m_arena) {
-	if (!core || !core->dbg || !core->dbg->maps) {
-		return false;
+
+	r_return_val_if_fail (core && core->dbg && core->dbg->maps, false);
+
+	if (core->dbg->main_arena_resolved) {
+		return true;
 	}
 
 	GHT brk_start = GHT_MAX, brk_end = GHT_MAX;
 	GHT libc_addr_sta = GHT_MAX, libc_addr_end = 0;
 	GHT addr_srch = GHT_MAX, heap_sz = GHT_MAX;
+	const char *libc_path = NULL;
+	GHT libc_addr = GHT_MAX;
+	GHT main_arena_sym = GHT_MAX;
+	bool is_debugged = true;
```

Comment:
I think you could do
```suggestion
	bool is_debugged = r_config_get_i (core->config, "cfg.debug");
```
and use `is_debugged` everywhere in this function instead of always getting the value from `r_config_get_i`.

Reply:
Make sense